core/frontend/src/RealityDataSource.ts

```diff
@@ -265,3 +285,87 @@ export class RealityDataSourceProviderRegistry {
     return this._providers.get(name);
   }
 }
+
+/**
+ * Options for creating a Google Photorealistic 3D Tiles (GP3DT) reality data source provider.
+ * @beta
+ */
+export interface RealityDataSourceGP3DTProviderOptions {
```

Comment:
The caller needs to provide *either* an API key *or* a `getAuthToken` function - never both and never neither - right?
Otherwise, e.g., your provider will produce URLs with `?key=undefined`.
Can you adjust this interface to require exactly one to be specified?

Reply:
Yes that's correct, just changed this to a union type that requires either one of these but not both.